lib/charty/backends/plotly.rb

```diff
@@ -0,0 +1,82 @@
+require 'json'
+require 'securerandom'
+
+module Charty
+  class Plotly < PlotterAdapter
+    Name = "plotly"
+    attr_reader :context
+
+    def initilize
+    end
+
+    def label(x, y)
+    end
+
+    def series=(series)
+      @series = series
+    end
+
+    def render(context, filename)
+      plot(nil, context)
+    end
+
+    def plot(plot, context)
+      context = context
+
+      case context.method
+      when :bar
+        render_graph(context, :bar)
+      when :curve
+        render_graph(context, :scatter)
+      when :scatter
+        render_graph(context, nil, options: {data: {mode: "markers"}})
+      else
+        raise NotImplementedError
+      end
+    end
+
+    private
+
+    def div_id
+      "charty-plotly-#{SecureRandom.uuid}"
+    end
+
+    def div_style
+      "width: 100%;height: 100%;"
+    end
+
+    def render_graph(context, type, options: {})
+      data = context.series.map do |series|
+        {
+          type: type,
+          x: series.xs.to_a,
+          y: series.ys.to_a,
+          name: series.label
+        }.merge(options[:data] || {})
+      end
+      layout = {
+        title: { text: context.title },
+        xaxis: {
+          title: context.xlabel,
+          range: [context.range[:x].first, context.range[:x].last]
+        },
+        yaxis: {
+          title: context.ylabel,
+          range: [context.range[:y].first, context.range[:y].last]
+        }
+      }
+      render_html(data, layout)
+    end
+
+    def render_html(data, layout)
+      id = div_id
+      <<~FRAGMENT
+        <script type='text/javascript' src="https://cdn.plot.ly/plotly-latest.min.js"></script>
```

Comment:
If I use this Plotly backend to develop a Web application, I would like to run `npm install plotly.js-dist` and use the library. There, I think it is better to leave install to the user only for JavaScript library more than writing script tag `src =" https://cdn.plot.ly/plotly-latest.min.js "` inside the Charty library. The convenience may be a bit lost, How do you think?

Reply:
Use `src="https://cdn.plot.ly/plotly-latest.min.js"` by default.
However, if we write our own header, for example, we want to be able to use it. I think this is good.

Reply:
Fixed in 334a77f
It is almost same as google_chart's load tag. https://github.com/red-data-tools/charty/blob/master/lib/charty/backends/google_chart.rb

Reply:
Thanks! I will merge it.